File: repositories/flag_repository.go
package repositories

import (
	"context"
	"time"

	"doctor-service/entities"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type FlagRepository interface {
	Create(ctx context.Context, flag *entities.VerificationFlag) error
	FindByDoctorID(ctx context.Context, doctorID uuid.UUID) ([]entities.VerificationFlag, error)
	ResolveFlag(ctx context.Context, flagID, adminID uuid.UUID) error
}

type GormFlagRepository struct {
	db *gorm.DB
}

func NewFlagRepository(db *gorm.DB) FlagRepository {
	return &GormFlagRepository{db: db}
}

func (r *GormFlagRepository) Create(ctx context.Context, flag *entities.VerificationFlag) error {
	return r.db.WithContext(ctx).Create(flag).Error
}

func (r *GormFlagRepository) FindByDoctorID(ctx context.Context, doctorID uuid.UUID) ([]entities.VerificationFlag, error) {
	var flags []entities.VerificationFlag
	err := r.db.WithContext(ctx).Where("doctor_id = ?", doctorID).Order("created_at DESC").Find(&flags).Error
	return flags, err
}

func (r *GormFlagRepository) ResolveFlag(ctx context.Context, flagID, adminID uuid.UUID) error {
	now := time.Now()
	return r.db.WithContext(ctx).Model(&entities.VerificationFlag{}).
		Where("id = ?", flagID).
		Updates(map[string]interface{}{
			"resolved":    true,
			"resolved_by": &adminID,
			"resolved_at": &now,
		}).Error
}

File: backend_api/repositories/flag_repository.py
import datetime
from typing import List
from uuid import UUID
from sqlalchemy.orm import Session
from entities.flag import VerificationFlag

class FlagRepository:
    def __init__(self, db: Session):
        self.db = db

    def Create(self, flag: VerificationFlag) -> None:
        self.db.add(flag)
        self.db.commit()
        self.db.refresh(flag)

    def FindByDoctorID(self, doctor_id: UUID) -> List[VerificationFlag]:
        return self.db.query(VerificationFlag).filter(VerificationFlag.doctor_id == doctor_id).order_by(VerificationFlag.created_at.desc()).all()

    def ResolveFlag(self, flag_id: UUID, admin_id: UUID) -> None:
        now = datetime.datetime.now()
        self.db.query(VerificationFlag).filter(VerificationFlag.id == flag_id).update({
            "resolved": True,
            "resolved_by": admin_id,
            "resolved_at": now
        })
        self.db.commit()
